### ISSUE TITLE
[RFC] New dependency: `react-native-image-picker` 

### ISSUE THREAD

**Issue by damassi** [artsy/eigen#3473]
#### New Dependency
Name: react-native-image-picker
URL: https://github.com/react-native-community/react-native-image-picker

### Focus

A React Native module that allows you to use native UI to select a photo/video from the device library or directly from the camera.

Check out [the example code](https://github.com/react-native-community/react-native-image-picker/blob/master/example/App.js) and a test out PR [here](https://github.com/artsy/eigen/pull/3474). 

#### Check List
- [x] Have read over the source code, and we can maintain it
- [x] Has had a release in the last year, or looks done and stable

#### Alternatives
`react-native-image-picker` seems to be the standard. There's also [react-native-cameraroll](https://github.com/react-native-community/react-native-cameraroll), which we already have in the app, but it is extremely low level and doesn't give us all of the nice native bits.

**Comment by MounirDhahri** [comment]
I think we should also consider trying [react-native-image-crop-picker ](https://github.com/ivpusic/react-native-image-crop-picker) since it supports multiple image selection and it's fairly stable and maintained as well. About react-native-cameraroll, I agree with what you said.

**Comment by ds300** [comment]
The existing picker we have in the current consignments flow provides a significantly better UX than either of these options IMO. Is there some functionality it's missing for the new flow?

**Comment by ashfurrow** [comment]
Yeah, I'd like to hear more about the specifics UX that's motivating the RFC. Maybe some mocks of what this will look like to users, in contrast to what we already have?

**Comment by damassi** [comment]
We'd like to go with native UI, versus custom UI.

**Comment by damassi** [comment]
@MounirDhahri - I did see that come up! The name seemed a bit misleading, but now that I'm thinking about it this could be beneficial for users when they take photo. Will dig in.

**Comment by ashfurrow** [comment]
Okay, let's circle back to this. 

I've reviewed [the new designs](https://www.figma.com/file/yDSHlbJVzBxvSUROD2bfRI/Sell-tab) around image picking for consignment submissions and I have some thoughts. The downside of the designs is that users need to choose between choosing an existing image or taking a new one (whereas our current implementation lets users do either/both without choosing upfront). 

![Screen Shot 2020-07-20 at 4 11 27 PM](https://user-images.githubusercontent.com/498212/87981716-b1d18600-caa3-11ea-95d1-c17b59fbe33e.png)

This is actually a limitation of the iOS native image picker API, but as a consequence, this means that most users are familiar with with this UX. So I'm okay with this tradeoff of using the native image picker.

That's the UX downside, but there are upsides too. First, our image picker only shows the album of all images, sorted by their date taken. This new design would let users select from user-created albums, or auto-generated albums like the user's favourites. That's cool! And I think serves a real-world use case.

![ios_multiple_pick_v2](https://user-images.githubusercontent.com/498212/87981942-07a62e00-caa4-11ea-8fdf-56e6855a94dd.png)

So let's move forward with the discussion assuming the designs are what we want to implement; let's evaluate this RFC based on how it helps us achieve our UX goals. The existing image-picking code for the consignments flow is... it's pretty bad 😅 To implement the new designs, we would basically need to throw it all away and start over. The code we would write (to access the iOS image library) wouldn't be unique or differentiating to our app, which makes it a good candidate for an OSS dependency. Using a dependency would also provide an abstraction over iOS/Android, so we have less work to do going forward. So from both a UX and a codebase health perspective, I'm a 👍 on using _a_ dependency.

However... `react-native-image-picker` does _not_ support the design UX, which has multiple image selection. I think we should go with Mounir's suggestion and use `react-native-image-crop-picker`. It provides a UI _similar_ to the native experience, but better. The `react-native-image-picker` README even calls this out:

> This library started as a basic bridge of the native iOS image picker, and I want to keep it that way. As such, functionality beyond what the native `UIImagePickerController` supports will not be supported here. **Multiple image selection, more control over the crop tool, and landscape support** are things missing from the native iOS functionality - **not issues with my library**. If you need these things, [react-native-image-crop-picker](https://github.com/ivpusic/react-native-image-crop-picker) might be a better choice for you.

Using `react-native-image-crop-picker` gets us a UX that the user is familiar with, but is improved in a way that I think suits our designed UX. I've looked over the Objective-C code and it's all pretty standard, and the React API seems straightforward.
